flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/test/java/org/apache/flink/cdc/connectors/polardbx/PolardbxSourceTestBase.java

```diff
@@ -113,14 +115,34 @@ protected static Connection getJdbcConnection() throws SQLException {
     }
 
     protected static Boolean checkConnection() {
-        LOG.info("check polardbx connection validation...");
-        try {
-            Connection connection = getJdbcConnection();
-            return connection.isValid(3);
+        LOG.info("check PolarDBx CDC node status...");
+        boolean cdcReady = false;
+        try (Connection connection = getJdbcConnection();
+                Statement statement = connection.createStatement()) {
+            ResultSet rs = statement.executeQuery("SHOW MASTER STATUS");
+            if (rs.next()) {
+                String binlogFilename = rs.getString(1);
+                long binlogPosition = rs.getLong(2);
+                // the initial binlog position is 4
+                cdcReady = StringUtils.isNotEmpty(binlogFilename) && binlogPosition > 4L;
+                if (cdcReady) {
+                    LOG.info(
+                            "PolarDBx CDC node is ready at offset {}:{}",
+                            binlogFilename,
+                            binlogPosition);
+                } else {
+                    LOG.warn(
+                            "PolarDBx CDC node is not ready at offset {}:{},  waiting...",
+                            binlogFilename,
+                            binlogPosition);
+                }
+            } else {
+                LOG.warn("PolarDBx CDC node is not ready, waiting...");
+            }
         } catch (SQLException e) {
-            LOG.warn("polardbx connection is not valid... caused by:" + e.getMessage());
-            return false;
+            LOG.warn("PolarDBx CDC node is not ready... caused by:{}", e.getMessage());
```

Comment:
It is recommended to set cdcReady to false in the catch block to prevent it from being overlooked after adding new logic.